### ISSUE TITLE
Helm chart for Errbit

### ISSUE THREAD

**Issue by tmaier** [errbit/errbit#1244]
I just created a [Helm chart](https://helm.sh) for Errbit.
Helm is something like a templating language and management engine for Kubernetes.
It allows to easily deploy and upgrade Errbit clusters (Errbit + MongoDB)

It is currently located at https://github.com/tmaier/errbit-helm

It would be great if people with Kubernetes and Errbit experience would give feedback on it. (e.g. @pigri, @jasquat or @burnettk)

I also added a docker-entrypoint script which constructs the `MONGO_URL` on the fly and delays the start of errbit when the mongodb database is not ready yet. See https://github.com/tmaier/errbit-helm/blob/master/dockerfile/docker-entrypoint.sh
It would be great if you could take this over to this repo to make the custom [tmaier/errbit_helm](https://hub.docker.com/r/tmaier/errbit-helm/) image obsolete.

I am also thinking about a more generic home for the Helm chart. One could be https://github.com/errbit and another could be https://github.com/kubernetes/charts.
The advantage would be a higher visibility towards the chart as well as proper CI/CD processes in place. See [Owning and Maintaining A Chart](https://github.com/kubernetes/charts#owning-and-maintaining-a-chart)

**Comment by burnettk** [comment]
I tried to fire it up but I'm not in an environment that supports PersistentVolumeClaim, so mongo can't start. Mongo has an option, `persistence.enabled` (https://github.com/kubernetes/charts/tree/master/stable/mongodb), that, if I could set it to false, I think would make it work (at least to test). I'm not certain how this sort of thing is generally handled. Maybe it would be possible for the chart to take a parameter that would be passed along to the dependent mongo chart? Maybe there's a better way?

https://github.com/kubernetes/charts seems like the awesome place for it once the kinks are worked out and it's production ready, and keeping it at https://github.com/tmaier/errbit-helm or in the errbit/errbit repo temporarily seems fine to me.

unrelated, this is the first thing i tried, since i wanted to avoid adding the repo, and i assumed i'd be able to install it from disk like i've done with other charts:

````
git clone git@github.com:tmaier/errbit-helm.git
helm install --name my-errbit ./errbit-helm
````

got:
````
Error: found in requirements.yaml, but missing in charts/ directory: mongodb
````

i guess running `helm repo add` per the README.md and installing it from the repo allows it to continue to grab mongodb from the other public repo.

**Comment by tmaier** [comment]
Hi,

thanks for trying it out.
I would recommend to leverage the inheritance feature of Helm. This means you have multiple `values.yaml` files. One within each Helm Chart (there is the main chart `errbit-helm` and its subchart `mongodb`) and then there is a local one you should maintain with your custom configuration.
The load order is 1) subcharts 2) chart 3) custom `values.yaml`. Later one would override the defaults of the others. This means you would create a `values.yaml` file where `persistence.enabeled` is `false`.

Regarding your error message. This is indeed due to the fact that you were cloning it. You need to do what `bundle install` does at ruby: You need to install the dependencies. Run `helm dependency build` to do so.

So in total, you should be able to run the following commands (untested)
```bash
echo "mongodb\n  persistence:\n    enabled: false" > values.yaml
git clone git@github.com:tmaier/errbit-helm.git
helm dependency build
helm install --name my-errbit ./errbit-helm
```

You could also run `helm install --name errbit --dry-run --debug ./errbit-helm > kubernetes.yaml`  to get the generated Kubernetes objects.

I just released version 0.3.0. It has two batch jobs. One to bootstrap errbit on `helm install` and one for `rake db:migrate` (run only on `helm upgrade`

**Comment by burnettk** [comment]
using the [inheritance stuff](https://github.com/kubernetes/helm/issues/630) largely worked, nice call! 

working four-liner with no mongo persistence:

````
echo "mongodb:\n  persistence:\n    enabled: false" > values.yaml
git clone git@github.com:tmaier/errbit-helm.git
helm dependency build errbit-helm
helm install -f values.yaml --name my-errbit ./errbit-helm
````

**Comment by jasquat** [comment]
@burnettk and I made some updates to the errbit fork at https://github.com/jasquat/errbit.git to get the health checks working with [errbit-helm](https://github.com/tmaier/errbit-helm). We also added the docker-entrypoint.sh entrypoint from errbit-helm to this fork. I'm not sure if the errbit repo is the correct place for it but seems appropriate.

We're still testing out this fork and we will submit a pull request once we have everything working.
